src/dolphin/timeseries.py

```diff
@@ -520,5 +592,76 @@ def read_and_solve(
         num_threads=num_threads,
     )
     writer.notify_finished()
-    # Return the output files
+
+    if add_overviews:
+        logger.info("Creating overviews for unwrapped images")
+        create_overviews(out_paths, image_type=ImageType.UNWRAPPED)
     return out_paths
+
+
+def correlation_to_variance(correlation: ArrayLike, nlooks: int) -> Array:
+    r"""Convert interferometric correlation to phase variance.
+
+    Uses the CRLB formula from Rodriguez, 1992 [1]_ to get the phase variance,
+    \sigma_{\phi}^2:
+
+    \[
+        \sigma_{\phi}^{2} = \frac{1}{2N_{L}} \frac{1 - \gamma^{2}}{\gamma^{2}}
+    \]
+
+    where \gamma is the correlation and N_L is the effective number of looks.
+
+    References
+    ----------
+    .. [1] Rodriguez, E., and J. M. Martin. "Theory and design of interferometric
+    synthetic aperture radars." IEEE Proceedings of Radar and Signal Processing.
+    Vol. 139. No. 2. IET Digital Library, 1992.
+
+    """
+    return (1 - correlation**2) / (2 * nlooks * correlation**2 + 1e-6)
+
+
+def select_reference_point(
+    ccl_file_list: Sequence[PathOrStr],
+    amp_dispersion_file: PathOrStr,
+    output_dir: Path,
+    block_shape: tuple[int, int] = (512, 512),
+    num_threads: int = 5,
+) -> ReferencePoint:
+    """Automatically select a reference point for a stack of unwrapped interferograms.
+
+    Uses the amplitude dispersion and connected component labels, the point is selected
+    which
+
+    1. is within intersection of all nonzero connected component labels (always valid)
+    2. has the lowest amplitude dispersion
```

Comment:
Can we ensure the point is within the largest connected component?

Reply:
v2 :)

Reply:
(I first wanted to see if there's a way to do it without adding `scikit-image` as a dependency, which is mintpy's method)

Reply:
Oh I think I found a way with scipy-

```python
from scipy import ndimage
from collections import counter

In [59]: labels, nlabel = ndimage.label(cci, structure=np.ones((3, 3)))

In [60]: %time Counter(labels.ravel()).most_common(10)
CPU times: user 250 ms, sys: 559 µs, total: 250 ms
Wall time: 249 ms
Out[60]: 

[
    (0, 4751432),
    (370, 45694),
    (1168, 1640),
    (729, 1366),
    (1280, 1001),
    (1110, 926),
    (1197, 817),
    (1005, 796),
    (304, 731),
    (407, 629)
]

```
so label 370 has the largest number of pixels

Reply:
sounds great